keps/sig-apps/592-ttl-after-finish/README.md

```diff
@@ -250,17 +257,141 @@ Mitigations:
+  No.
+
+* **Can the feature be disabled once it has been enabled (i.e. can we roll back
+  the enablement)?**
+  Yes.
+
+* **What happens if we reenable the feature if it was previously rolled back?**
+  It should work as expected.
+
+* **Are there any tests for feature enablement/disablement?**
+  No.
+
+### Rollout, Upgrade and Rollback Planning
+
+* **How can a rollout fail? Can it impact already running workloads?**
+  It shouldn't impact already running workloads. This is an opt-in feature since users need to 
+  explicitly set the TTLSecondsAfterFinished parameter in the job spec, which is accepted by the 
+  api-server only if the feature is enabled.
+
+* **What specific metrics should inform a rollback?**
+  Unexpected restarts of kube-controller-manager
+
+* **Were upgrade and rollback tested? Was the upgrade->downgrade->upgrade path tested?**
+  N/A
+
+* **Is the rollout accompanied by any deprecations and/or removals of features, APIs, 
+fields of API types, flags, etc.?**
+  No
+
+### Monitoring Requirements
+
+_This section must be completed when targeting beta graduation to a release._
+
+* **How can an operator determine if the feature is in use by workloads?**
+  - Check the workqueue metrics
+  - When checking the job object in the api server, the TTLSecondsAfterFinished parameter should 
+    be set if it was specified when the job was created.
+
+* **What are the SLIs (Service Level Indicators) an operator can use to determine 
+the health of the service?**
+ - [x] Metrics
+    - Metric name: `ttl_after_finished_controller_rate_limiter_use`
+    - Metric name: `workqueue_adds_total`
+    - Metric name: `workqueue_depth`
+    - Metric name: `workqueue_queue_duration_seconds`
+    - Metric name: `workqueue_retries_total`
+    - Components exposing the metric: `kube-controller-manager`
+    - Metric name: `etcd_object_counts{resource="jobs.batch"}`
+    - Components exposing the metric: `kube-apiserver`.
+
+* **What are the reasonable SLOs (Service Level Objectives) for the above SLIs?**
+
+99% of the jobs that needs cleanup are deleted within X minutes.
 
+* **Are there any missing metrics that would be useful to have to improve observability 
+of this feature?**
+
+No
+
+### Dependencies
+
+_This section must be completed when targeting beta graduation to a release._
+
+* **Does this feature depend on any specific services running in the cluster?**
+  No.
+
+### Scalability
+
+* **Will enabling / using this feature result in any new API calls?**
+  - API call type: DELETE jobs
+  - Estimated throughput: depends on job creation and completion rate.
+  - originating component(s): kube-controller-manager
+
+* **Will enabling / using this feature result in introducing new API types?**
+  No.
+  
+* **Will enabling / using this feature result in any new calls to the cloud 
+provider?**
+ No.
+
+* **Will enabling / using this feature result in increasing size or count of 
+the existing API objects?**
+ No.
```

Comment:
Technically yes - we're adding a field so the object will be slightly bigger :)

Reply:
updated